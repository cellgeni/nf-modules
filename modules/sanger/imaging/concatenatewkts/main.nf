process IMAGING_CONCATENATEWKTS {
    tag "${meta.id}"
    label 'process_single'

    // conda "${moduleDir}/environment.yml"
    container "${workflow.containerEngine == 'singularity' && !task.ext.singularity_pull_docker_container
        ? 'quay.io/bioinfotongli/tiled_spotiflow:0.5.2'
        : 'quay.io/bioinfotongli/tiled_spotiflow:0.5.2'}"

    input:
    tuple val(meta), path(wkts)

    output:
    tuple val(meta), path("${output_name}"), emit: concatenated_peaks
    path "versions.yml", emit: versions

    when:
    task.ext.when == null || task.ext.when

    script:
    def args = task.ext.args ?: ''
    def prefix = task.ext.prefix ?: "${meta.id}"
    output_name = "${prefix}_merged_peaks.csv"
    """
    merge_wkts.py run \\
        -output_name ${output_name} \\
        ${wkts} \\
        ${args}

    cat <<-END_VERSIONS > versions.yml
    "${task.process}":
        imaging: \$(merge_wkts.py version)
    END_VERSIONS
    """

    stub:
    def prefix = task.ext.prefix ?: "${meta.id}"
    output_name = "${prefix}_merged_peaks.csv"
    """
    touch ${output_name}

    cat <<-END_VERSIONS > versions.yml
    "${task.process}":
        imaging: \$(merge_wkts.py version)
    END_VERSIONS
    """
}
