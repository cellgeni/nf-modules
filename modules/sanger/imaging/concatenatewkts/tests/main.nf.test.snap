{
    "mouse heart - wkt ": {
        "content": [
            {
                "0": [
                    [
                        {
                            "id": "test"
                        },
                        "test_merged_peaks.csv:md5,d5e0dfaa5058349ba1eb51fe9b286b58"
                    ]
                ],
                "1": [
                    "versions.yml:md5,fdcbf4adb208c7d07a3288f0634f3610"
                ],
                "concatenated_peaks": [
                    [
                        {
                            "id": "test"
                        },
                        "test_merged_peaks.csv:md5,d5e0dfaa5058349ba1eb51fe9b286b58"
                    ]
                ],
                "versions": [
                    "versions.yml:md5,fdcbf4adb208c7d07a3288f0634f3610"
                ]
            }
        ],
        "meta": {
            "nf-test": "0.9.2",
            "nextflow": "24.04.4"
        },
        "timestamp": "2025-02-07T14:58:56.032796301"
    },
    "mouse heart - wkt - stub": {
        "content": [
            {
                "0": [
                    [
                        {
                            "id": "test"
                        },
                        "test_merged_peaks.csv:md5,d41d8cd98f00b204e9800998ecf8427e"
                    ]
                ],
                "1": [
                    "versions.yml:md5,fdcbf4adb208c7d07a3288f0634f3610"
                ],
                "concatenated_peaks": [
                    [
                        {
                            "id": "test"
                        },
                        "test_merged_peaks.csv:md5,d41d8cd98f00b204e9800998ecf8427e"
                    ]
                ],
                "versions": [
                    "versions.yml:md5,fdcbf4adb208c7d07a3288f0634f3610"
                ]
            }
        ],
        "meta": {
            "nf-test": "0.9.2",
            "nextflow": "24.04.4"
        },
        "timestamp": "2025-02-07T14:59:03.601526069"
    }
}